app/controllers/api/v01/bibs_controller.rb

```diff
@@ -29,28 +29,44 @@ def create_or_update_teacher_sets
       end
 
       teacher_set = TeacherSet.where(bnumber: "b#{bnumber}").first_or_initialize
-      teacher_set.update_attributes(
-        title: title,
-        call_number: var_field('091'),
-        description: var_field('520'),
-        edition: var_field('250'),
-        isbn: var_field('020'),
-        primary_language: fixed_field('24'),
-        publisher: var_field('260'),
-        contents: var_field('505'),
-        primary_subject: var_field('690', false),
-        physical_description: physical_description,
-        details_url: "http://catalog.nypl.org/record=b#{teacher_set_record['id']}~S1",
-        grade_begin: grade_or_lexile_array('grade')[0],
-        grade_end: grade_or_lexile_array('grade')[1],
-        lexile_begin: grade_or_lexile_array('lexile')[0],
-        lexile_end: grade_or_lexile_array('lexile')[1],
-        availability: 'available',
-        available_copies: 999
-      )
-      teacher_set.update_subjects_via_api(all_var_fields('650', 'a'))
-      teacher_set.update_notes(var_field('500', true))
-      teacher_set.update_included_book_list(teacher_set_record)
+      begin
+        teacher_set.update_attributes(
+          title: title,
+          call_number: var_field('091'),
+          description: var_field('520'),
+          edition: var_field('250'),
+          isbn: var_field('020'),
+          primary_language: fixed_field('24'),
+          publisher: var_field('260'),
+          contents: var_field('505'),
+          primary_subject: var_field('690', false),
+          physical_description: physical_description,
+          details_url: "http://catalog.nypl.org/record=b#{teacher_set_record['id']}~S1",
+          grade_begin: grade_or_lexile_array('grade')[0],
+          grade_end: grade_or_lexile_array('grade')[1],
+          lexile_begin: grade_or_lexile_array('lexile')[0],
+          lexile_end: grade_or_lexile_array('lexile')[1],
+          availability: 'available',
+          available_copies: 999
+        )
+      rescue => e
+        AdminMailer.failed_bibs_controller_api_request(@request_body, "One attribute may be too long.  Error: #{e.message[0..200]}...", action_name, teacher_set).deliver
```

Comment:
@daryanypl I can see how they wouldn't want that, but they are library staff (rather than patrons) and the library would benefit if they learned better ways to manage their emails.

Is there another option?  Can you check with Courtney about it and let us know what you'd like to do?

Reply:
@daryanypl fortunately this will only send one email and then the lambda moves on to the next record.